src/lib/tls/tls_session.h

```diff
@@ -288,6 +288,8 @@ class BOTAN_PUBLIC_API(3, 0) Session_Summary : public Session_Base {
 
       std::string kex_algo() const { return m_kex_algo; }
 
+      std::optional<std::string> kex_parameters() const { return m_kex_parameters; }
```

Comment:
```suggestion
      const std::optional<std::string>& kex_parameters() const { return m_kex_parameters; }
```

... there might even be some `clang-tidy` hiccup about that, I think.